### PR TITLE
Merge actions when extending AbstractBehavior

### PR DESCRIPTION
**Pull request by sesquipedalian-dev**
- when extending an AbstractBehavior, run all of the actions defined for both the abstract behavior and the concrete behavior
- to allow customization when needed in extending, add an 'order' field to determine the order the actions are run

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,6 +142,28 @@ Actions are a series of functions to run. Availabe actions are:
...
+        topic: hello_kafka_out
+        payload: >
+          {
+            "kafka": "OK",
+            "data": {}
+          }
+    - sleep:
+        duration: 1s
+      # sleep first
+      order: -1
```
Comment (zhouzhuojie) [inline README.md]: I like the order field, very neat. One comment about having `-1` is that in a lot of languages it means the last one, can we do something different here?

Comment (sesquipedalian-dev) [reply README.md]: option 1: if I set this value `-1000` is that clearer? 
option 2: should we only allow 0 and positive integers for order?  my thinking was it's conceptually difficult to change the order of any actions you inherited with extends, so it makes more sense to let users go before the 'default' value.   Maybe if we use positive integers the default value should be 100.

File: model.go [hunk 2]
```diff
@@ -114,6 +114,7 @@ type (
 
 // Action represents actions
 type ActionDispatcher struct {
+	Order              int                `yaml:"order"`
```
[the remark this answers is not in the document]
Comment (sesquipedalian-dev) [reply model.go]: I'll fix this

File: README.md (hunk 1 above)
Comment (sesquipedalian-dev) [reply README.md]: pushed a version with the examples showing `-1000` for clarity

File: README.md [hunk 3]
```diff
@@ -176,7 +198,8 @@ Templates can be useful to assemble your payloads from parts
 ```
 
 ### Abstract Behaviors
-Abstract Behaviors can be used to parameterize some data
+Abstract Behaviors can be used to parameterize some data.
+When an abstract behavior and a behavior extending it both have actions defined, all of them are run when the behavior matches.
```
Comment (zhouzhuojie) [inline README.md]: Can you add a paragraph of how actions are merged and ordered?

Comment (zhouzhuojie) [reply README.md]: for example, I can see that actions are deduplicated, and order by `order asc`

Comment (sesquipedalian-dev) [reply README.md]:
```
The actions by default run in the order defined in the mock file; you can adjust this by adding an int 'order' value. The default value for 'order' is 0.
```

```
When an abstract behavior and a behavior extending it both have actions defined, all of them are run when the behavior matches.  Actions will run in order of the 'order' field; if this is the same for two actions the action defined earlier in the abstract behavior runs first, followed by actions in the concrete behavior.
```

Comment (sesquipedalian-dev) [reply README.md]: Also, are they deduplicated? I don't think so.  For example you may run multiple 'sleep' if you want.

File: load.go [hunk 4]
```diff
@@ -107,9 +108,28 @@ func (om *OpenMock) populateBehaviors(mocks []*Mock) {
...
+					Actions: orderedActions,
+				})
+				r.Behaviors[m.Key] = m
+			}
+		}
 	}
 }
 
+func actionsEqual(lhs []ActionDispatcher, rhs []ActionDispatcher) bool {
+	lhsString, lhsError := yaml.Marshal(lhs)
```
Comment (zhouzhuojie) [inline load.go]: does yaml marshal have consistent ordering for map type?

File: README.md (hunk 3 above)
Comment (zhouzhuojie) [reply README.md]: misread that `assertEqual` function, yup, no deduplication

File: load.go (hunk 4 above)
Comment (sesquipedalian-dev) [reply load.go]: I believe yes: https://blog.labix.org/2014/09/22/announcing-yaml-v2-for-go#mapslice

**Pull request merged**